outputs.tf

```diff
@@ -143,3 +143,11 @@ output "ebs_csi_account" {
 output "k8s_version" {
   value = data.aws_eks_cluster.cluster.version
 }
+
+output "aws_shared_credentials" {
+  value = local.aws_shared_credentials
+  precondition {
+    condition = length(var.aws_shared_credentials_file) == 0 || var.aws_shared_credentials_files == null
```

Comment:
Seems like you're missing the null check on the aws_shared_credentials_file and the length check on the aws_shared_credentials_files

Reply:
This precondition check is only intended to alert customers if they happened to configure a value for both of those input variable at the same time. The additional checks that you mention are not required in order to be able to do that given what the defaults are for those variables.